components/auth/org.wso2.carbon.auth.user.mgt/src/test/java/org/wso2/carbon/auth/user/mgt/internal/UserStoreMgtComponentTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *  * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied.  See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package org.wso2.carbon.auth.user.mgt.internal;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.osgi.framework.BundleContext;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.wso2.carbon.auth.user.mgt.UserStoreManager;
+import org.wso2.carbon.auth.user.mgt.UserStoreManagerFactory;
+import org.wso2.carbon.auth.user.store.configuration.UserStoreConfigurationService;
+import org.wso2.carbon.auth.user.store.configuration.models.UserStoreConfiguration;
+import org.wso2.carbon.auth.user.store.constant.UserStoreConstants;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({UserStoreManagerFactory.class})
+public class UserStoreMgtComponentTest {
+
+    @Test
+    public void testActivate() throws Exception {
+
+        PowerMockito.mockStatic(UserStoreManagerFactory.class);
```

Comment:
We should not use powermock. Let's try to write test cases without powermock.

Reply:
what is our suggestion to mock the static classes?

Reply:
Mocking static methods should be avoided at all costs because there is no clean way of mocking them. Either the static method must be tested in an integration test(without mocking), or the static method should be written in a way that it does not require mocking(a functional style method that returns an output based on an input only). Static methods that require mocking in order to enable unit testing is a sign that the code is no longer testable and needs to be re-factored. Power mockito exists in order to help us work with legacy code that cannot be re-factored.

Reply:
We have decided not to use Powermock in C5 development. AFAIR Powermock is not supported for JDK 9,10 atm. Also, it seems they are not actively developing Powermock. https://github.com/powermock/powermock/issues/901

Reply:
I believe carbon-apimgt repo extensively used power mock in c5 branch. We need to do a complete restucture to avoid using power mock. I'm okey to remove power mock but we shouldn't introduce spcial code like we did in c4 to support testing.

Reply:
@uvindra has removed some of the Powermock code and also refactored the code as well. We are not accepting any PR which introduces Powermock.